### PR TITLE
Support Iter8 v0.1.0 

### PR DESCRIPTION
**Pull request by jadeyliu939**
** Describe the change **

1. Adding additional iter8 experiment info ( assessments and criteria) for Iter8tExperimentGet() rest call to support New Iter8 Experiment Detail page
2. Change handling of StartedTime,CreatedTime and EndedTime ( was string, now is int64)
3. Support Iter8 v0.1

_Please describe what the code change is about_

** Issue reference **
#2633 

Please reference the original [pull request 2729](https://github.com/kiali/kiali/pull/2729)

### CODE REVIEW

**Comment by rhqci** [comment]
Can one of the admins verify this patch?

File: business/iter8.go [hunk 1]
```diff
@@ -170,6 +171,35 @@ func (in *Iter8Service) ParseJsonForCreate(body []byte) (string, error) {
 	object.Spec.TrafficControl.MaxTrafficPercentage = newExperimentSpec.TrafficControl.MaxTrafficPercentage
 	object.Spec.TrafficControl.MaxIterations = newExperimentSpec.TrafficControl.MaxIterations
 	object.Spec.TrafficControl.TrafficStepSize = newExperimentSpec.TrafficControl.TrafficStepSize
+	object.Spec.Analysis.AnalyticsService = "http://iter8-analytics.iter8:" + strconv.Itoa(in.k8s.GetAnalyticPort())
```
Comment (lucasponce) [inline business/iter8.go:174]: Is this `object.Spec.Analysis.AnalyticsService` used just for informational usecases with the user ?
Or is there any planned query trom Kiali to this service ?
Also this url seems perhaps it needs to be in the config.go (Iter8 section) and probably in the operator as defaults.
I can take care of the operator to simplify the work.

Comment (lucasponce) [reply business/iter8.go:174]: Ok, I can respond to myself, so, it's just to show this info to the user as data but no logic involved.

**Comment by lucasponce** [comment]
Also some compiler errors:
```
[lponce@newrecopolis kiali]$ make test
Running tests, excluding third party tests under vendor
go test github.com/kiali/kiali github.com/kiali/kiali/appstate github.com/kiali/kiali/business github.com/kiali/kiali/business/checkers github.com/kiali/kiali/business/checkers/authorization github.com/kiali/kiali/business/checkers/destinationrules github.com/kiali/kiali/business/checkers/gateways github.com/kiali/kiali/business/checkers/meshpolicies github.com/kiali/kiali/business/checkers/policies github.com/kiali/kiali/business/checkers/serviceentries github.com/kiali/kiali/business/checkers/services github.com/kiali/kiali/business/checkers/sidecars github.com/kiali/kiali/business/checkers/virtual_services github.com/kiali/kiali/config github.com/kiali/kiali/config/security github.com/kiali/kiali/graph github.com/kiali/kiali/graph/api github.com/kiali/kiali/graph/config/cytoscape github.com/kiali/kiali/graph/telemetry github.com/kiali/kiali/graph/telemetry/istio github.com/kiali/kiali/graph/telemetry/istio/appender github.com/kiali/kiali/graph/telemetry/istio/util github.com/kiali/kiali/handlers github.com/kiali/kiali/jaeger github.com/kiali/kiali/jaeger/jaegertest github.com/kiali/kiali/kubernetes github.com/kiali/kiali/kubernetes/cache github.com/kiali/kiali/kubernetes/kubetest github.com/kiali/kiali/ldap github.com/kiali/kiali/log github.com/kiali/kiali/models github.com/kiali/kiali/prometheus github.com/kiali/kiali/prometheus/internalmetrics github.com/kiali/kiali/prometheus/prometheustest github.com/kiali/kiali/routing github.com/kiali/kiali/server github.com/kiali/kiali/status github.com/kiali/kiali/tests/data github.com/kiali/kiali/util github.com/kiali/kiali/util/httputil github.com/kiali/kiali/util/intutil
# github.com/kiali/kiali/business [github.com/kiali/kiali/business.test]
business/apps_test.go:22:26: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to NewWithBackends:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
business/apps_test.go:23:20: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in field value:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
business/health_test.go:34:22: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in field value:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
business/health_test.go:34:74: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to NewWithBackends:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
business/health_test.go:59:22: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in field value:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
business/health_test.go:59:74: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to NewWithBackends:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
business/health_test.go:90:22: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in field value:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
business/health_test.go:90:74: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to NewWithBackends:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
business/health_test.go:119:22: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in field value:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
business/health_test.go:119:74: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to NewWithBackends:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
business/health_test.go:119:74: too many errors
# github.com/kiali/kiali/graph/api [github.com/kiali/kiali/graph/api.test]
graph/api/api_test.go:80:55: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to kubetest.NewK8SClientFactoryMock:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
graph/api/api_test.go:153:55: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to kubetest.NewK8SClientFactoryMock:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
# github.com/kiali/kiali/graph/telemetry/istio/appender [github.com/kiali/kiali/graph/telemetry/istio/appender.test]
graph/telemetry/istio/appender/dead_node_test.go:105:43: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to "github.com/kiali/kiali/business".NewWithBackends:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
graph/telemetry/istio/appender/istio_details_test.go:69:43: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to "github.com/kiali/kiali/business".NewWithBackends:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
graph/telemetry/istio/appender/istio_details_test.go:131:43: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to "github.com/kiali/kiali/business".NewWithBackends:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
graph/telemetry/istio/appender/istio_details_test.go:200:43: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to "github.com/kiali/kiali/business".NewWithBackends:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
graph/telemetry/istio/appender/service_entry_test.go:52:43: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to "github.com/kiali/kiali/business".NewWithBackends:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
graph/telemetry/istio/appender/service_entry_test.go:257:43: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to "github.com/kiali/kiali/business".NewWithBackends:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
graph/telemetry/istio/appender/sidecars_check_test.go:216:43: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to "github.com/kiali/kiali/business".NewWithBackends:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
# github.com/kiali/kiali/handlers [github.com/kiali/kiali/handlers.test]
handlers/apps_test.go:170:55: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to kubetest.NewK8SClientFactoryMock:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
handlers/apps_test.go:183:55: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to kubetest.NewK8SClientFactoryMock:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
handlers/health_test.go:61:55: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to kubetest.NewK8SClientFactoryMock:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
handlers/health_test.go:112:55: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to kubetest.NewK8SClientFactoryMock:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
handlers/health_test.go:156:55: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to kubetest.NewK8SClientFactoryMock:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
handlers/namespaces_test.go:210:55: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to kubetest.NewK8SClientFactoryMock:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
handlers/services_test.go:349:55: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to kubetest.NewK8SClientFactoryMock:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
handlers/workloads_test.go:35:55: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to kubetest.NewK8SClientFactoryMock:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
handlers/workloads_test.go:334:55: cannot use k8s (type *kubetest.K8SClientMock) as type "github.com/kiali/kiali/kubernetes".IstioClientInterface in argument to kubetest.NewK8SClientFactoryMock:
        *kubetest.K8SClientMock does not implement "github.com/kiali/kiali/kubernetes".IstioClientInterface (missing GetAnalyticPort method)
ok      github.com/kiali/kiali  0.086s
?       github.com/kiali/kiali/appstate [no test files]
FAIL    github.com/kiali/kiali/business [build failed]
ok      github.com/kiali/kiali/business/checkers        0.065s
ok      github.com/kiali/kiali/business/checkers/authorization  0.034s
ok      github.com/kiali/kiali/business/checkers/destinationrules       0.112s
ok      github.com/kiali/kiali/business/checkers/gateways       0.043s
ok      github.com/kiali/kiali/business/checkers/meshpolicies   0.059s
ok      github.com/kiali/kiali/business/checkers/policies       0.043s
ok      github.com/kiali/kiali/business/checkers/serviceentries 0.060s
ok      github.com/kiali/kiali/business/checkers/services       0.062s
ok      github.com/kiali/kiali/business/checkers/sidecars       0.028s
ok      github.com/kiali/kiali/business/checkers/virtual_services       0.041s
ok      github.com/kiali/kiali/config   0.013s
ok      github.com/kiali/kiali/config/security  (cached)
?       github.com/kiali/kiali/graph    [no test files]
FAIL    github.com/kiali/kiali/graph/api [build failed]
ok      github.com/kiali/kiali/graph/config/cytoscape   0.011s
?       github.com/kiali/kiali/graph/telemetry  [no test files]
?       github.com/kiali/kiali/graph/telemetry/istio    [no test files]
FAIL    github.com/kiali/kiali/graph/telemetry/istio/appender [build failed]
?       github.com/kiali/kiali/graph/telemetry/istio/util       [no test files]
FAIL    github.com/kiali/kiali/handlers [build failed]
?       github.com/kiali/kiali/jaeger   [no test files]
?       github.com/kiali/kiali/jaeger/jaegertest        [no test files]
ok      github.com/kiali/kiali/kubernetes       0.322s
ok      github.com/kiali/kiali/kubernetes/cache 0.009s
?       github.com/kiali/kiali/kubernetes/kubetest      [no test files]
?       github.com/kiali/kiali/ldap     [no test files]
?       github.com/kiali/kiali/log      [no test files]
ok      github.com/kiali/kiali/models   0.011s
?       github.com/kiali/kiali/prometheus       [no test files]
ok      github.com/kiali/kiali/prometheus/internalmetrics       (cached)
ok      github.com/kiali/kiali/prometheus/prometheustest        0.365s
ok      github.com/kiali/kiali/routing  0.011s
ok      github.com/kiali/kiali/server   0.285s
ok      github.com/kiali/kiali/status   0.109s
?       github.com/kiali/kiali/tests/data       [no test files]
ok      github.com/kiali/kiali/util     (cached)
?       github.com/kiali/kiali/util/httputil    [no test files]
?       github.com/kiali/kiali/util/intutil     [no test files]
FAIL
make: *** [make/Makefile.build.mk:55: test] Error 2
```

This is because we have modified the Iter8 interface:
```
type Iter8ClientInterface interface {
	CreateIter8Experiment(namespace string, json string) (Iter8Experiment, error)
	DeleteIter8Experiment(namespace string, name string) error
	GetIter8Experiment(namespace string, name string) (Iter8Experiment, error)
	GetIter8Experiments(namespace string) ([]Iter8Experiment, error)
	IsIter8Api() bool
	Iter8ConfigMap() ([]string, error)
	GetAnalyticPort() int
}
```

Which is part of the `IstioClientInterface` so, it needs a mock test under the kubetest/mock.go.
That would solve these issues I think.

Let ke know if you can update it, if not I can try to prepare a PRs for it, too.

**Pull request merged**